### ISSUE TITLE
What happens if I delete my webid card?

### ISSUE THREAD

**Issue by dmitrizagidulin** [solid/solid#27]
Should we allow users to delete their own WebID profile card? (They won't be able to log in or access the account, if that happens, so this would essentially be deleting their account).

Given that we intend to have a separate [API for deleting accounts](https://github.com/solid/solid/issues/14), how do we prevent users from deleting their profile? Do we intercept the request, and redirect them to the start of the Delete Account workflow? Do we just not allow it, via ACL?

(moved from https://github.com/solid/solid-spec/issues/51)

**Comment by bblfish** [comment]
The user could delete everything in the card, or just his public keys and still have the same problem.  

There a few solutions to this problem:

 * the owner of the cloud service could keep the e-mail address and other information in a separate resource owned him and not allow that to be changed. As the owner would also have access to some mode of payment of the user that should be kept there too. That info could be used to fix that problem.
 * In any case it would help to have a history for each resource which would allow the administrator to come back to a previous setting.

**Comment by ebporter** [comment]
Some thoughts on these questions:
1) Given that we intend to have a separate API for deleting accounts, how do we prevent users from deleting their profile?
I think we should prevent a user from deleting their profile -- either mistakenly or intentionally -- and for those who are eager to suppress their public view, instead offer them an opportunity to suppress it or deactivate it or hide it (which I think just means putting all their profile data into the Preferences file and not making it Public).
I'm not sure what use case gave rise to the idea of deleting a profile, but from a usability and user support point of view, it's better to shelve/archive/hide the data than to delete it, unless there's a darn good reason.

2) Do we intercept the request, and redirect them to the start of the Delete Account workflow?
If they just want to hide it from view, give them a workflow for that. If they actually want to delete their account, push them to that workflow.

3) Do we just not allow it, via ACL?
I'm suggesting we don't allow it.

**Comment by Mitzi-Laszlo** [comment]
You can find more discussion on this subject here https://forum.solidproject.org/t/what-happens-if-i-delete-my-webid-card/382

**Comment by bourgeoa** [comment]
On top of the ACL way which I feel not fully safe, there could be an account initial backup preferably outside of /root pod.
The user can have access on the server page to an admin Pod page where with his credentials ( checked against the .dB/users/<Pod record>) he could have access to an API with copy/rebuild of his profile card and other Pod admin 
 possibilities (delete, change external WebID, pod user name, password, security email, backup extended profile .....).

The server page is then a Pod admin/creation page.

**Comment by csarven** [comment]
Overall, I don't really like the idea of a special API to delete accounts. It should all happen through the common interface.

>Should we allow users to delete their own WebID profile card? (They won't be able to log in or access the account, if that happens, so this would essentially be deleting their account).

I think there are generally two cases:

* WebID is coupled with a user account.
* WebID is decoupled from a user account.

The former (what you're asking I think) is a typical use case of asking the service/storage/id provider to delete their account. It is all within their right, I think; at least applicable under some jurisdictions following certain regulations. At the very least, I think Solid should enable such possibility.

Potential mistakes or issues around that is a separate issue and can be mitigated if for instance the system prompts the requester with "are you super really sure? there is no turning back.. export your stuff.." . If the WebID also happens to be the first/root/admin user on that system, there can be stricter or higher requirements before the request goes through.

For the latter case, I think there is more of a separation between the WebID Profile, IDP, Storage and so on. I don't see why it shouldn't be possible to delete in this case given fair warnings.

As to what happens once the delete request is approved eg:
* give the requester a chance to export/backup stuff
* (if applicable) move user home dir to trashbin (so maybe "undo" would be possible if we need to), return 404/410 (that's a separate discussion - github issues around that)
* update ACLs
* perhaps prompt to send a notification to all their contacts/followers so that they can remove their references if they like
* update groups
* deluser

**Comment by bblfish** [comment]
One could also have a server that keeps the history of the resources, so that mistakes can be undone. There would be a lot of use cases for this. Then somone will ask for there to be a DELETE functionality that removes all history.

In the end that comes down to keeping backups of one's pod, so as to restore it to a previous state in case  of grave error.

**Comment by csarven** [comment]
Interesting to note that LDP uses non-normative text to mention altering resources: https://www.w3.org/TR/ldp/#ldp-http-method-side-effects

>6.2.4 LDP servers can alter the state of other resources as a result of any HTTP request, especially when non-safe methods are used ([RFC7231] section 4.2.1). For example, it is acceptable for the server to remove triples from other resources whose subject or object is the deleted resource as the result of a successful HTTP DELETE request. It is also acceptable and common for LDP servers to not do this – the server's behavior can vary, so LDP clients cannot depend on it.
